lua/astrocommunity/utility/toggleterm-manager/init.lua

```diff
@@ -0,0 +1,50 @@
+return {
+  {
+    "ryanmsnyder/toggleterm-manager.nvim",
+    opts = function(_, opts)
+      local toggleterm_manager = require "toggleterm-manager"
+      local actions = toggleterm_manager.actions
+
+      local mappings = {
+        i = {
+          ["<CR>"] = { action = actions.toggle_term, exit_on_action = true }, -- toggles terminal open/closed
+          ["<C-i>"] = { action = actions.create_term, exit_on_action = false }, -- creates a new terminal buffer
+          ["<C-d>"] = { action = actions.delete_term, exit_on_action = false }, -- deletes a terminal buffer
+          ["<C-r>"] = { action = actions.rename_term, exit_on_action = false }, -- provides a prompt to rename a terminal
+        },
+        n = {
+          ["<CR>"] = { action = actions.toggle_term, exit_on_action = true }, -- toggles terminal open/closed
+          ["r"] = { action = actions.rename_term, exit_on_action = false }, -- provides a prompt to rename a terminal
+          ["d"] = { action = actions.delete_term, exit_on_action = false }, -- deletes a terminal buffer
+          ["n"] = { action = actions.create_term, exit_on_action = false }, -- creates a new terminal buffer
+        },
+      }
+
+      if not opts.mappings then
+        opts.mappings = mappings
+      else
+        opts.mappings = vim.tbl_deep_extend("force", opts.mappings, mappings)
+      end
```

Comment:
```suggestion
      opts.mappings = require("astrocore").extend_tbl(opts.mappings, {
        i = {
          ["<CR>"] = { action = actions.toggle_term, exit_on_action = true }, -- toggles terminal open/closed
          ["<C-i>"] = { action = actions.create_term, exit_on_action = false }, -- creates a new terminal buffer
          ["<C-d>"] = { action = actions.delete_term, exit_on_action = false }, -- deletes a terminal buffer
          ["<C-r>"] = { action = actions.rename_term, exit_on_action = false }, -- provides a prompt to rename a terminal
        },
        n = {
          ["<CR>"] = { action = actions.toggle_term, exit_on_action = true }, -- toggles terminal open/closed
          ["r"] = { action = actions.rename_term, exit_on_action = false }, -- provides a prompt to rename a terminal
          ["d"] = { action = actions.delete_term, exit_on_action = false }, -- deletes a terminal buffer
          ["n"] = { action = actions.create_term, exit_on_action = false }, -- creates a new terminal buffer
        },
      })
```

We can utilize the `extend_tbl` in AstroCore to simplify the code which adds the checks you are wanting.

Reply:
cool! nice to learn!